extension/observer/dockerobserver/integration_test.go

```diff
@@ -58,21 +58,71 @@ func TestObserverEmitsEndpointsIntegration(t *testing.T) {
 	c := container.New(t)
 	image := "docker.io/library/nginx:1.17"
 	cntr := c.StartImage(image, container.WithPortReady(80))
+	config := NewFactory().CreateDefaultConfig().(*Config)
+	config.CacheSyncInterval = 1 * time.Second
+	config.UseHostBindings = true
+	config.UseHostnameIfPresent = true
+	mn := &mockNotifier{endpointsMap: map[observer.EndpointID]observer.Endpoint{}}
+	obvs := startObserverWithConfig(t, mn, config)
+	time.Sleep(2 * time.Second) // wait for endpoints to sync
```

Comment:
Would `require.Eventually()` potentially reduce flake?

Reply:
Refactored all these to use Eventually when I could